### PR TITLE
[stable10] Webdav TrashBin API

### PR DESCRIPTION
**Pull request by DeepDiver1975**
backport #35636

### CODE REVIEW

**Review by individual-it — APPROVED** [review]

File: apps/dav/lib/TrashBin/TrashBinPlugin.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+		$propFind->handle(self::TRASHBIN_ORIGINAL_LOCATION, static function () use ($node) {
+			return $node->getOriginalLocation();
+		});
+
+		$propFind->handle(self::TRASHBIN_DELETE_TIMESTAMP, static function () use ($node) {
+			return $node->getDeleteTimestamp();
+		});
+
+		$propFind->handle(self::TRASHBIN_DELETE_DATETIME, static function () use ($node) {
```
Comment (butonic) [inline apps/dav/lib/TrashBin/TrashBinPlugin.php]: I think we can leave out datetime. the delete timestamp by definition (= unix timestamp) is in the GMT timezone ... or at least should be. and everything I know can digest unix timestamps. we can add it, but should recommend using the timestamp.

**Review by butonic — APPROVED** [review]

File: apps/dav/lib/TrashBin/TrashBinPlugin.php (hunk 1 above)
Comment (DeepDiver1975) [reply apps/dav/lib/TrashBin/TrashBinPlugin.php]: unix timestamps will limit us to the interval of unix timestamps 01.01.1970-19.1.2038(int32 based)
Using a real date time format is not limited to this. 

furthermore the timezone is fixed/not part of the transmission - using a real date time format can support this (even if not used as of today)

Comment (DeepDiver1975) [reply apps/dav/lib/TrashBin/TrashBinPlugin.php]: and as discussed: http://apiux.com/2013/03/20/5-laws-api-dates-and-times/ advised to use ISO-8601 - we use http://tools.ietf.org/html/rfc7231#section-7.1.1.1 to stay consistent with the getlastmodified property

Comment (butonic) [reply apps/dav/lib/TrashBin/TrashBinPlugin.php]: that makes sense ... nevertheless, unix timestamps are supposed to be GMT / UTC but that is jist screaming for errors ;-)

**Pull request merged**